### PR TITLE
JAMES-3647 Adopt eclipse-temurin:11-jre-focal docker image

### PR DESCRIPTION
**Pull request by vttranlina**
Some update of  https://github.com/apache/james-project/pull/647

### CODE REVIEW

File: src/site/Dockerfile [hunk 1]
```diff
@@ -2,7 +2,7 @@
 #
 # VERSION	1.0
 
-FROM adoptopenjdk:11-jdk-hotspot
+FROM eclipse-temurin:11-jre-focal
```
Comment (chibenwa) [inline src/site/Dockerfile:5]: We IMO need a JDK to run maven?

**Review by chibenwa — COMMENTED** [review]
Good catches, thanks!

File: examples/custom-james-assembly/pom.xml [hunk 2]
```diff
@@ -54,7 +54,7 @@
                 <version>2.7.1</version>
                 <configuration>
                     <from>
-                        <image>adoptopenjdk:11-jdk-hotspot</image>
+                        <image>eclipse-temurin:11-jre-focal</image>
```
Comment (Arsnael) [inline examples/custom-james-assembly/pom.xml:57]: jdk too here

Comment (chibenwa) [reply examples/custom-james-assembly/pom.xml:57]: No JRE is fine here

File: src/site/Dockerfile (hunk 1 above)
Comment (vttranlina) [reply src/site/Dockerfile:5]: You're right,

Comment (vttranlina) [reply src/site/Dockerfile:5]: Screenshot from 2021-09-10 12-53-04

Comment (vttranlina) [reply src/site/Dockerfile:5]:
```suggestion
FROM adoptopenjdk:11-jdk-hotspot
```

Comment (chibenwa) [reply src/site/Dockerfile]: Can't we use `eclipse-temurin:11-jdk-focal` ?

Comment (vttranlina) [reply src/site/Dockerfile]: updated `eclipse-temurin:11-jdk-focal`

**Review by chibenwa — APPROVED** [review]

**Review by Arsnael — APPROVED** [review]

**Pull request merged**